src/Microsoft.Data.SqlClient/tests/tools/TDS/TDS.EndPoint/TDSServerParser.cs

```diff
@@ -69,15 +69,19 @@ public void Run()
                 {
                     case TDSMessageType.PreLogin:
                         {
-                            if (Session.Encryption == TDSEncryptionType.None)
+                            if ((Session.Encryption == TDSEncryptionType.None || Session.Encryption == TDSEncryptionType.Off) &&
+                                (MessageBeingReceived[0] as TDSPreLoginToken).Encryption != TDSPreLoginTokenEncryptionType.On)
                             {
-                                (MessageBeingReceived[0] as TDSPreLoginToken).Encryption = TDSPreLoginTokenEncryptionType.None;
+                                if (Session.Encryption == TDSEncryptionType.None)
+                                    (MessageBeingReceived[0] as TDSPreLoginToken).Encryption = TDSPreLoginTokenEncryptionType.None;
+                                else
+                                    (MessageBeingReceived[0] as TDSPreLoginToken).Encryption = TDSPreLoginTokenEncryptionType.Off;
                             }
 
                             // Call into the subscriber to process the packet
                             responseMessages = Server.OnPreLoginRequest(Session, MessageBeingReceived);
 
-                            if (Session.Encryption == TDSEncryptionType.None)
+                            if (Session.Encryption == TDSEncryptionType.None || Session.Encryption == TDSEncryptionType.Off)
```

Comment:
I'm not sure about these changes. Even the previous lines from 72-75, I think they might be unnecessary. I think any logic changes here should go in Server.OnPreLoginRequest. I haven't had time to debug this fully, so let me know if you disagree.

Reply:
When a TestTdsServer is created, in the GenericTDSServerSession constructor, the default Encryption = TDSEncryptionType.Off. However, in TDSServerParser.cs class, the Session.Encryption = TDSEncryptionType.Off value was not handled and so, MessageBeingReceived.Encryption and the transport layer encryption are both not getting disabled.

Should I change the default value of Session.Encryption from TDSEncryptionType.Off to TDSEncryptionType.None instead so no changes are necessary for TDSServerParser.cs file?

Reply:
The unnecessary changes were reverted.